src/renderers/dom/ReactDOM.js

```diff
@@ -133,6 +133,13 @@ if (__DEV__) {
       }
     }
   }
+
+  // Check if using Polymer and we are not using native shadow
+  var shadyDom = window.Polymer !== undefined && !window.Polymer.useNativeShadow;
```

Comment:
> We would need to check that window exists here, as it may not on the server, for example. I think usually ExecutionEnvironment.canUseDOM is the check we use (you can search for it in the source).

Noted. Will update.

> I would prefer we check != null rather than !== undefined here in the case where there is a Polymer variable set to null so member access doesn't throw later. I know this is unlikely but still.

Mmmm. Good point.

> Should we emit the warning if Polymer.useShadow is false? I'm not sure what it means (will need to read some docs) but from the code it seems like useNativeShadow would then also become false regardless of the polyfill.

Yeah, the documentation is pretty sparse on this, but your Googling skills might be better than mine. It looks like `useShadow` checks to see if the user passed a `{ dom: 'shadow' }` property when configuring Polymer and if they have a native shadow polyfill. I think what we actually want to check for is `Polymer.settings.wantShadow && !Polymer.settings.nativeShadow` to check if they are requesting a shadow but must rely on the shady dom.